core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -83,10 +84,16 @@ public static Channel intercept(Channel channel, ClientInterceptor... intercepto
    */
   public static Channel intercept(Channel channel, List<? extends ClientInterceptor> interceptors) {
     Preconditions.checkNotNull(channel, "channel");
+    Channel newChannel = channel;
     for (ClientInterceptor interceptor : interceptors) {
-      channel = new InterceptorChannel(channel, interceptor);
+      newChannel = new InterceptorChannel(newChannel, interceptor);
+    }
+
+    if (channel instanceof ManagedChannel) {
```

Comment:
A ManagedChannel overload _may_ be better. Although it is sort of nice we're not advertising this too much in this current implementation. Hmm...